soccer/gameplay/plays/restarts/our_indirect_pivot.py

```diff
@@ -0,0 +1,153 @@
+    class State(enum.Enum):
+        passing = 1
+        kicking = 2
+
+    MinChipRange = 0.3
+    MaxChipRange = 3.0
+    ChipperPower = 1
+    TargetSegmentWidth = 1.5
+    MaxKickSpeed = 1
+    MaxKickAccel = 1
+    Running = False
+
+    def __init__(self, indirect=None):
+        super().__init__(continuous=True)
+
+        # setup a line kick skill to replace the pivotkick since a pivot would easily cause a double touch
+        self.kicker = skills.pivot_kick.PivotKick()
+        self.kicker.chip_power = OurIndirectPivot.ChipperPower  # TODO: base this on the target dist from the bot
+        self.kicker.min_chip_range = OurIndirectPivot.MinChipRange
+        self.kicker.max_chip_range = OurIndirectPivot.MaxChipRange
+        self.kicker.max_speed = OurIndirectPivot.MaxKickSpeed
+        self.kicker.max_accel = OurIndirectPivot.MaxKickAccel
+
+        # larger avoid ball radius for line kick setup so we don't run over the ball backwards
+        #self.kicker.setup_ball_avoid = constants.Field.CenterRadius - constants.Robot.Radius
+        #self.kicker.drive_around_dist = constants.Field.CenterRadius - constants.Robot.Radius
+
+        # create a one touch pass behavior with line kick as the skill
+        self.pass_bhvr = tactics.coordinated_pass.CoordinatedPass()
+
+        self.add_state(OurIndirectPivot.State.passing,
+                       behavior.Behavior.State.running)
+        self.add_state(OurIndirectPivot.State.kicking,
+                       behavior.Behavior.State.running)
+
+        # add transistions for when the play is done
+        self.add_transition(behavior.Behavior.State.start,
+                            OurIndirectPivot.State.passing, lambda: True,
+                            'immediately')
+
+        self.add_transition(OurIndirectPivot.State.passing,
+                            OurIndirectPivot.State.kicking, lambda: self.
+                            pass_bhvr.is_done_running(), 'Pass Completed')
+
+        self.add_transition(OurIndirectPivot.State.kicking,
+                            behavior.Behavior.State.completed, lambda: False,
+                            'Shot Completed')
+
+    @classmethod
+    def score(cls):
+        gs = main.game_state()
+
+        # enter play when doing a corner kick or stay in it even if we manipulate the ball
+        if OurIndirectPivot.Running or (
+                gs.is_ready_state() and
+            (gs.is_our_direct() or gs.is_our_indirect()) and
+                main.ball().pos.y < (constants.Field.Length - 1.2)):
+            OurIndirectPivot.Running = True
+            return 0
+        else:
+            return float("inf")
+
+    def on_enter_passing(self):
+        # start the actual pass
+        self.add_subbehavior(self.pass_bhvr, 'pass')
+        self.pass_bhvr.receive_point = self.pick_pass_spot()
+        self.pass_bhvr.use_chipper = self.evaluate_chip(self.pick_pass_spot())
+
+        if abs(main.ball().pos.x) > 2:
```

Comment:
This logic for generating the threat or rebound point seems to be duplicated, maybe it would be nicer to have it in a helper function, and/or define these positions on the field as constants

Reply:
This is one of the ones that is getting fixed up as well